Source/Core/Core/PowerPC/JitCommon/JitCache.h

```diff
@@ -203,7 +206,16 @@ class JitBaseBlockCache
   // It is used to provide a fast way to query if no icache invalidation is needed.
   ValidBlockBitSet valid_block;
 
-  // This array is indexed with the masked PC and likely holds the correct block id.
+  // This array is indexed with the shifted PC and likely holds the correct block id.
   // This is used as a fast cache of block_map used in the assembly dispatcher.
-  std::array<JitBlock*, FAST_BLOCK_MAP_ELEMENTS> fast_block_map{};  // start_addr & mask -> number
+  // It is implemented via a shm segment using m_block_map_arena.
+  JitBlock** fast_block_map = 0;
+  Common::MemArena m_block_map_arena;
+
+  // An alternative for the above fast_block_map but without a shm segment
+  // in case the shm memory region couldn't be allocated.
+  std::array<JitBlock*, FAST_BLOCK_MAP_FALLBACK_ELEMENTS>
+      fast_block_map_fallback{};  // start_addr & mask -> number
+
+  JitBlock** fast_block_map_ptr = 0;
```

Comment:
If you're already touching these, can you attach a `m_` to the front of the names?